facts/bigsur_upgrade_supported.py

```diff
@@ -0,0 +1,251 @@
+    return NSString.alloc().initWithData_encoding_(
+        raw_value, NSUTF8StringEncoding
+    ).rstrip('\0')
+
+
+def sysctl(name, output_type=str):
+    '''Wrapper for sysctl so we don't have to use subprocess'''
+    size = c_uint(0)
+    # Find out how big our buffer will be
+    libc.sysctlbyname(name, None, byref(size), None, 0)
+    # Make the buffer
+    buf = create_string_buffer(size.value)
+    # Re-run, but provide the buffer
+    libc.sysctlbyname(name, buf, byref(size), None, 0)
+    if output_type in (str, 'str'):
+        return buf.value.decode('UTF-8')
+    if output_type in (int, 'int'):
+        # complex stuff to cast the buffer contents to a Python int
+        if size.value == 4:
+            return cast(buf, POINTER(c_int32)).contents.value
+        if size.value == 8:
+            return cast(buf, POINTER(c_int64)).contents.value
+    if output_type == 'raw':
+        # sysctl can also return a 'struct' type; just return the raw buffer
+        return buf.raw
+
+
+def is_virtual_machine():
+    '''Returns True if this is a VM, False otherwise'''
+    cpu_features = sysctl('machdep.cpu.features').split()
+    return 'VMM' in cpu_features
+
+
+def is_supported_model():
+    '''Returns False if model is in list of unsupported models,
+    True otherwise'''
+    non_supported_models = [
+        u'MacBook10,1',
+        u'MacBook8,1',
+        u'MacBook9,1',
+        u'MacBookAir6,1',
+        u'MacBookAir6,2',
+        u'MacBookAir7,1',
+        u'MacBookAir7,2',
+        u'MacBookAir8,1',
+        u'MacBookAir8,2',
+        u'MacBookPro11,1',
+        u'MacBookPro11,2',
+        u'MacBookPro11,3',
+        u'MacBookPro11,4',
+        u'MacBookPro11,5',
+        u'MacBookPro12,1',
+        u'MacBookPro13,1',
+        u'MacBookPro13,2',
+        u'MacBookPro13,3',
+        u'MacBookPro14,1',
+        u'MacBookPro14,2',
+        u'MacBookPro14,3',
+        u'MacBookPro15,1',
+        u'MacBookPro15,2',
+        u'MacBookPro15,3',
+        u'MacBookPro15,4',
+        u'MacPro6,1',
+        u'MacPro7,1',
+        u'Macmini7,1',
+        u'Macmini8,1',
+        u'iMac14,4',
+        u'iMac15,1',
+        u'iMac16,1',
+        u'iMac16,2',
+        u'iMac17,1',
+        u'iMac18,1',
+        u'iMac18,2',
+        u'iMac18,3',
+        u'iMac19,1',
+        u'iMac19,2',
+        u'iMacPro1,1',
+    ]
+    current_model = get_current_model()
+    if not current_model or current_model in non_supported_models:
+        return False
+    else:
+        return True
+
+
+def get_minor_system_version():
+    '''Returns 7 for Lion, 8 for Mountain Lion, etc'''
+    darwin_version = int(os.uname()[2].split('.')[0])
+    return darwin_version - 4
+
+
+def is_supported_system_version():
+    '''Returns True if current macOS version is 10.7 through 10.13,
+    False otherwise'''
+    macos_minor_version = get_minor_system_version()
+    if macos_minor_version >= 16:
+        return False
+    elif macos_minor_version >= 7:
```

Comment:
According to the [Big Sur](https://apps.apple.com/us/app/macos-big-sur/id1526878132?mt=12) listing in the App Store, the minimum supported version should be 10.9, not 10.7.